draft-irtf-cfrg-spake2.xml

```diff
@@ -103,23 +104,28 @@
       </section>
     </section>
     <section title="Table of points for common groups">
-      <t>Every curve presented in the table below has an OID from <xref
-        target="RFC5480" />. We construct a string using the OID and the needed
-        constant, for instance "1.3.132.0.35 point generation seed (M)" for
-        P-512.  This string is turned into an infinite sequence of bytes by
-        hashing with SHA256, and hashing that output again to generate the next
-        32 bytes, and so on. This pattern is repeated for each group and value,
-        with the string modified appropriately.</t>
-
-      <t>The initial segment of bytes of length equal to that of an encoded
-        group element is taken, and is then formatted as required for the
-        group.  In the case of Weierstrass points, this means setting the first
-        byte to 0x02 or 0x03 depending on the low-order bit. For Ed25519 style
-        formats this means taking all the bytes as the representation of the
-        group element.  This string of bytes is then interpreted as a point in
-        the group. If this is impossible, then the next non-overlapping segment
-        of sufficient length is taken. We multiply that point by the cofactor
-        h, and if that is not the identity, output it.</t>
+      <t>For each curve in the table below, we construct a string
+      using the curve OID from <xref target="RFC5480" /> or its name,
+      combined with the needed constant, for instance "1.3.132.0.35
+      point generation seed (M)" for P-512.  This string is turned
+      into a series of blocks by hashing with SHA256, and hashing that
+      output again to generate the next 32 bytes, and so on.  This
+      pattern is repeated for each group and value, with the string
+      modified appropriately.</t>
+
+      <t>A byte string of length equal to that of an encoded group
+      element is constructed by concatenating as many blocks as are
+      required, starting from the first block, and truncating to the
+      desired length.  The byte string is then formatted as required
+      for the group.  In the case of Weierstrass curves, this means
+      setting the first byte to 0x02 or 0x03 depending on the
+      low-order bit.  For <xref target="RFC7748" /> curves this means
+      taking all the bytes as the representation of the group element.
+      The byte string is then interpreted as an element in the group.
+      If this interpretation yields a valid group element with the
+      correct order (p), the byte string is the output.  Otherwise,
+      the initial hash block is discarded and the process is repeated
+      until a valid element is found.</t>
```

Comment:
There is probably a bit more clarification possible here, but I think I can do that as a follow-up.